commit 4025fcee1cd58e87b7c5129a9a2ead8833f8443f
Author: Steve Muir <muir@fb.com>
Date:   Wed Jul 27 14:23:44 2016 -0700

    yummy:patchX:tests--don-t-run-private-device-tests-if-running-in-a-container.patch

tests: don't run private device tests if running in a container

    
    Private devices don't exist when running in a container, so skip the related
    tests.

diff --git a/src/test/test-execute.c b/src/test/test-execute.c
index 77ef4e8..baf5b96 100644
--- a/src/test/test-execute.c
+++ b/src/test/test-execute.c
@@ -33,6 +33,7 @@
 #include "test-helper.h"
 #include "unit.h"
 #include "util.h"
+#include "virt.h"
 
 typedef void (*test_function_t)(Manager *m);
 
@@ -111,6 +112,10 @@ static void test_exec_privatetmp(Manager *m) {
 }
 
 static void test_exec_privatedevices(Manager *m) {
+        if (detect_container() > 0) {
+                log_notice("testing in container, skipping private device tests");
+                return;
+        }
         test(m, "exec-privatedevices-yes.service", 0, CLD_EXITED);
         test(m, "exec-privatedevices-no.service", 0, CLD_EXITED);
 }
